worker/src/worker.py

```diff
@@ -121,22 +121,27 @@ def execute_case(case):
             step_item = workflow['blocks'][step]
 
             if step_item['type'] == 'action':
-                result = execute_block(case, step_item, step)
+                result = execute_block(case, step_item, step, was_suspended)
+
+                # Delete any state information from the block, as we no longer need it
+                if 'state' in case:
+                    del case['state']
 
                 if not result:
                     return
 
                 if result['type'] == 'result':
-                    save_result(case, result, step_item)
+                    if 'save_outputs' in step_item:
+                        save_result(case, result, step_item)
```

Comment:
I know we talked about storing only variables we explicitly are asked to store, however I think it may be better to store everything for explanatory reasons. We want each case to be transparent and reviewable.